src/transforms/parallel.jl

```diff
@@ -53,6 +53,9 @@ function apply(p::ParallelTableTransform, table)
   feats = first.(splits)
   metas = last.(splits)
 
+  # check the number of rows of generated tables
+  @assert allequal(_nrows(f) for f in feats) "parallel branches must produce the same number of rows"
```

Comment:
From the `@assert` docstring:

> An assert might be disabled at various optimization levels.

So, `@assert` should only be used for debugging. For actual interface behavior you should `throw` an exception.